cmd/crates/soroban-spec-tools/src/contract.rs

```diff
@@ -52,7 +52,13 @@ impl Spec {
                     "contractspecv0" => &mut spec,
                     _ => continue,
                 };
-                *out = Some(section.data());
+
+                if let Some(existing_data) = out.as_mut() {
+                    let combined_data = [existing_data, section.data()].concat();
+                    *out = Some(Box::leak(combined_data.into_boxed_slice()));
```

Comment:
We should avoid the use of the leak here, as this memory will never be recollected iiuc. We can make the meta a `Vec<u8>` so that we can concat and simply store a Vec. Would that work?

Reply:
Ah, good catch! It looks like changing `meta` to a `Vec<u8>` should work! I've also changed `env_meta` and `spec` as well, so we can continue handling the payload sections similarly. I get it pushed up shortly.